### ISSUE TITLE
SQLite to MySQL migration fails due to field order

### ISSUE THREAD

**Issue by kosli** [kevinpapst/kimai2#2484]
As I started on the docker version with kimai it had used the sqlite backend and now i want to migrate to a new installation with MySQL as I saw the post about SQLite not being supported in the future: https://www.kimai.org/blog/2021/sqlite-and-ftp-support-removed/
Based on the above link i created a dump of the SQLite file and grepped only the INSERTs, whereas I removed the INSERTs into the migrations table as these are already being created by `kimai:install`. I use Kimai 1.13 on both (docker & new prod environment). When I try to import the file into SQL it complains about `#1048 - Column 'password' cannot be null` for the `kimai2_users` entries.
The order for the fields in the SQLite file is different than in the newly created MySQL table. As the INSERT do not include the field names, the import fails. 
Despite the fact that I am puzzled that the order of the fields is different, I am looking into a way to create a dump of the sqlite DB which would include the field names in the INSERT statement 
Currently I am trying the different approaches mentioned here: https://stackoverflow.com/questions/4199850/sqlite-export-with-column-names.

@kevinpapst, any idea what could have caused in the first place the not matching order of fields?

**Comment by kevinpapst** [comment]
Probably the SQLite engine, which causes one bug after another... sorry for the troubles that might cause!

**Comment by kevinpapst** [comment]
BTW: I will update the blog post, becuase I forgot to mention it ... 

This tool generates dumps with full inserts (including column names): https://sqlitebrowser.org

**Comment by kevinpapst** [comment]
Blog post updated, please let me know if that helped with the migration

**Comment by kosli** [comment]
Thank you very much @kevinpapst
I haven't seen sqlitebrowser, but I had used in the meantime the script from @sixtyfive that you had mentioned in the blog too. My Kimai instance was successfully migrated that way.
FYI, on Ubuntu I had to install the packages `ruby-sequel ruby-mysql2 ruby-sqlite3 `and change in the script `Sequel.mysql` to `Sequel.mysql2`.
(Linking this issue to #1524)

**Comment by kevinpapst** [comment]
Thanks for the update @kosli - I updated the blog post with that info!

**Comment by kosli** [comment]
Further feedback:
For a 2nd kimai instance i tested the SQLite browser approach but it didn't work for me. 
Maybe it is related to MariaDB as mysql server, but from the sql file generated by SQLite browser I had to remove the "BEGIN TRANSACTION and COMMIT" commands and i had to remove the double quotation marks, otherwise the file failed during import. 
After the import I was not able to login with the original user/password, wheras the users table looked "okay". 
I cleared the database and imported via the Rakefile and that way it worked fine.
